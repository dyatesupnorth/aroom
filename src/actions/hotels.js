import axios from "axios";
axios.defaults.baseURL = 'http://localhost:3001/'
export const setHotels = hotels => ({
  type: "SET_HOTELS",
  hotels
});

export const startSetHotels = () => {
  return dispatch => {
    return axios.get("hotels").then(res => 
      dispatch(setHotels(res.data))
      );
  };
};
