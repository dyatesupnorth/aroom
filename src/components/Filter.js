import React, { Component } from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  setStarRatingFilter,
  setFacilitiesFilter
} from "../actions/filters";

export class Filter extends Component {
  
  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  };

  onStarRatingChange = e => {
    this.props.setStarRatingFilter(e.target.value);
  };

  onFacilitiesChange = (e, i) => {
    this.props.setFacilitiesFilter(e.target.value);
  };

  renderRadioButtons(maxStarRating) {
    // TODO: Pull max star rating dervied from max of all hotels
    // TODO: Convert to checkboxes, we want to be able to filter based on multiple selections
    let radioButtons = [];
    for (let i = 0; i < maxStarRating; i++) {
      radioButtons.push(
        <span key={i}>
          <input
            type="radio"
            value={i + 1}
            name="starRating"
            onChange={this.onStarRatingChange}
          />
          {i + 1}
        </span>
      );
    }
    return radioButtons;
  }

  renderCheckBoxes() {
    // TODO: Pull all available facilities directly from available hotels.
    return this.props.availableFacilities.map((facility, i) => (
      <label key={i}>
        {facility}
        <input
          name={facility}
          value={facility}
          type="checkbox"
          onChange={this.onFacilitiesChange}
        />
      </label>
    ));
  }

  render() {
    return (
      this.props.filters && (
        <div style={{ border: "1px solid tomato" }}>
          <h1>Filter</h1>
          <div className="form-group">
            <label>Name</label>
            <input
              className="form-control"
              name="text"
              type="text"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="form-group">
            <label>Star Rating</label>
            {this.renderRadioButtons(5)}
          </div>
          <div className="form-group">
            <label>Facilities</label>
            {this.renderCheckBoxes()}
          </div>
        </div>
      )
    );
  }
}

const mapStateToProps = state => {
  console.log("â€‹state", state);
  return {
    filters: state.filters,
    // Trying to get too fancy!
    availableFacilities: state.hotels.reduce(
      (acc, hotel) => (
        [].push.apply(acc, hotel.facilities.filter(v => acc.indexOf(v) === -1)),
        acc
      ),
      []
    )
  };
};

const mapDispatchToProps = dispatch => {
  return {
    setTextFilter: text => dispatch(setTextFilter(text)),
    setStarRatingFilter: starRating =>
      dispatch(setStarRatingFilter(starRating)),
    setFacilitiesFilter: facility => dispatch(setFacilitiesFilter(facility))
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Filter);
