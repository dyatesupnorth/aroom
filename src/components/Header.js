import React, { Component } from "react";

export class Header extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>Header</h1>
      </div>
    );
  }
}

export default Header;
