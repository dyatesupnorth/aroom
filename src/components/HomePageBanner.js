import React, { Component } from "react";

export class HomePageBanner extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>HomePageBanner</h1>
      </div>
    );
  }
}

export default HomePageBanner;
