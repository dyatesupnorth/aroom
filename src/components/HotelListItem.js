import React from "react";
import ListGroup from "react-bootstrap/ListGroup";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";
export const renderStars = noOfStars => {
	console.log("​noOfStars", noOfStars)
  let stars = [];
	
  for (let i = 0; i < noOfStars; i++) {
		console.log("​i", i)
    stars.push(<FontAwesomeIcon icon={faStar} />);
  }
  console.log("​stars", stars)
  return stars;
};
export const HotelListItem = props => {
  return (
    <ListGroup.Item className="list-group-item-action flex-column align-items-start">
      <div className="d-flex w-100 justify-content-between">
        <h5 className="mb-1">{props.name}</h5>
      </div>
      <p className="mb-1">{renderStars(props.starRating)}</p>
      <small>
        {!props.facilities || props.facilities.length === 0 ? (
          ''
        ) : (
          <span>
            <ul className="list-inline">
              {props.facilities.map(facility => {
                return (
                  <li className="list-inline-item" key={facility}>
                    {facility}
                  </li>
                );
              })}
            </ul>
          </span>
        )}
      </small>
    </ListGroup.Item>
  );
};

export default HotelListItem;
