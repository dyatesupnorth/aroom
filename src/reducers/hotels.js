const hotelsReducerDefaultState = [];
export default (state = hotelsReducerDefaultState, action) => {
  switch (action.type) {
    case "SET_EXPENSES":
      return action.hotels;
    default:
      return state;
  }
};
