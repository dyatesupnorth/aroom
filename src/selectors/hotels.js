export default (hotels, { text, starRating, facilities }) => {
  return hotels.filter(hotel => {
    const textMatch = hotel.name.toLowerCase().includes(text.toLowerCase());
    const starRatingMatch = starRating ? hotel.starRating >= starRating : true;
    const facilitiesMatch = facilities && facilities.length > 0 ? facilities.every(facility => hotel.facilities.includes(facility)) :true
    return textMatch && starRatingMatch && facilitiesMatch;
  });
};
