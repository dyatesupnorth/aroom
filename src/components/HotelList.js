import React from "react";
import { connect } from "react-redux";
import selectHotels from "../selectors/hotels";
import { HotelListItem } from "./HotelListItem";
import Card from "react-bootstrap/Card";

import ListGroup from "react-bootstrap/ListGroup";

export const HotelList = props => {
  return (
    <Card>
      <ListGroup variant="flush">
        {props.hotels.length === 0 ? (
          <p>No hotels</p>
        ) : (
          props.hotels.map(hotel => {
            return <HotelListItem key={hotel.name} {...hotel} />;
          })
        )}
      </ListGroup>
    </Card>
  );
};

const mapStateToProps = state => {
  return {
    hotels: selectHotels(state.hotels, state.filters)
  };
};
export default connect(
  mapStateToProps,
  undefined
)(HotelList);
