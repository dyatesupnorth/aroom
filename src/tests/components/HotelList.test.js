import React from "react";
import { shallow } from "enzyme";
import { HotelList } from "../../components/HotelList";
import hotels from "../fixtures/hotels";

test("should render HotelList correctly", () => {
  const wrapper = shallow(<HotelList hotels={hotels} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render HotelList with no hotels correctly", () => {
  const wrapper = shallow(<HotelList hotels={[]} />);
  expect(wrapper).toMatchSnapshot();
});


