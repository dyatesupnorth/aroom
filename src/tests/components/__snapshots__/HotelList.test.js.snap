// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HotelList correctly 1`] = `
<Card
  as="div"
  body={false}
>
  <ForwardRef
    variant="flush"
  >
    <HotelListItem
      facilities={
        Array [
          "car park",
          "pool",
        ]
      }
      key="hotelone"
      name="hotelone"
      starRating={5}
    />
    <HotelListItem
      facilities={
        Array [
          "car park",
          "gym",
        ]
      }
      key="hoteltwo"
      name="hoteltwo"
      starRating={3}
    />
    <HotelListItem
      facilities={Array []}
      key="hotelthree"
      name="hotelthree"
      starRating={3}
    />
  </ForwardRef>
</Card>
`;

exports[`should render HotelList with no hotels correctly 1`] = `
<Card
  as="div"
  body={false}
>
  <ForwardRef
    variant="flush"
  >
    <p>
      No hotels
    </p>
  </ForwardRef>
</Card>
`;
